File: 0792-binary-search/0792-binary-search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        low=0
        n=len(nums)
        high=n-1
        while low<=high:
            mid=(low+high)/2
            if nums[mid]==target:
                return mid
            elif nums[mid]<target:
                low=mid+1
            elif nums[mid]>target:
                high=mid-1
        return -1
        

File: 0792-binary-search/0792-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
     int m=nums.length;
      int s=0;

     
     int e=m-1;
     while(s<=e){
        int mid=(s+e)/2;
        if(nums[mid]==target){
            return mid;
        }
        else if(nums[mid]<target){
              s=mid+1;
        }
        else if(nums[mid]>target){
            e=mid-1;
        }
     }
     return -1;
        
    }
}